draft-packages/tag/KaizenDraft/Tag/Tag.tsx

```diff
@@ -103,6 +118,13 @@ const Tag = (props: TagProps) => {
                 </span>
               )
             }
+            if (variant === "profile") {
+              return (
+                <span className={styles.validationIcon}>
+                  {avatar && renderAvatar(avatar)}
```

Comment:
I'm a TypeScript noob, but I think there is a way of making props depending on each other, as in requiring the `avatar` prop when `variant` is `"profile"`.
This would mean someone using the `Tag` component with `variant="profile"` would get immediate feedback if they didn't provide the `avatar` prop.
This would then also make this check here (`avatar && ...`) obsolete.

Reply:
I've given it a crack using "Discriminative Unions" to achieve this. Seems to work :)